### PR TITLE
Add support for Cloud Watch Alarm Tags

### PR DESCRIPTION
**Pull request by gswai123**
This enhancement uses the augment function to add Tags to CloudWatch Alarm resources in Custodian, allowing users to filter by Alarm Tags using the Value Filter. The PR also includes a "Tag" action for CloudWatch alarms which allows users to add tags to alarm resources.

Issue Link: https://github.com/cloud-custodian/cloud-custodian/issues/6454

### CODE REVIEW

**Review by kapilt — APPROVED** [review]
lgtm, thanks

File: c7n/resources/cw.py [hunk 1]
```diff
@@ -37,6 +45,35 @@ class resource_type(TypeInfo):
         cfn_type = config_type = 'AWS::CloudWatch::Alarm'
 
     retry = staticmethod(get_retry(('Throttled',)))
+    source_mapping = {
+        'describe': DescribeAlarm,
+        'config': ConfigSource
+    }
+
+
+@Alarm.action_registry.register('tag')
```
Comment (PratMis) [inline c7n/resources/cw.py:54]: I think we may also need remove-tag, mark-for-op and marked-for-op. Would it be better to just use universal_taggble?

**Comment by kapilt** [comment]
Thanks that’s a good call, to get all the tag filter/actions for free so to
speak.

File: c7n/resources/cw.py [hunk 2]
```diff
@@ -34,7 +34,9 @@ class resource_type(TypeInfo):
         filter_type = 'list'
         name = 'AlarmName'
         date = 'AlarmConfigurationUpdatedTimestamp'
-        cfn_type = config_type = 'AWS::CloudWatch::Alarm'
+        universal_taggable = object()
```
Comment (PratMis) [inline c7n/resources/cw.py:37]: Please put back `cfn_type = 'AWS::CloudWatch::Alarm'`

Comment (kapilt) [reply c7n/resources/cw.py:38]: please restore the entire line including config_type

**Comment by PratMis** [comment]
Only thing I can see here is that `Tags` are not getting captured when `source: config`. AWS claims to be capturing the tags re: config cmdb(https://github.com/awslabs/aws-config-resource-schema/blob/e7586bdfd24f31427ccc8df21d760670bbcd1d93/config/properties/resource-types/AWS::CloudWatch::Alarm.properties.json), but it doesn't show up in reality

Response captured with config as source:
```
{'resourceId': 'c7n-test-alarm-tags-filter', 'configuration': {'Statistic': 'Average', 'Period': 3600.0, 'MetricName': 'CPUUtilization', 'ComparisonOperator': 'GreaterThanThreshold', 'AlarmConfigurationUpdatedTimestamp': datetime.datetime(2021, 3, 24, 18, 1, 32, 583000, tzinfo=tzutc()), 'AlarmName': 'c7n-test-alarm-tags-filter', 'Threshold': 10.0, 'Okactions': [], 'InsufficientDataActions': [], 'EvaluationPeriods': 5.0, 'ActionsEnabled': True, 'Namespace': 'AWS/EC2', 'Metrics': [], 'AlarmActions': [], 'AlarmArn': 'arn:aws:cloudwatch:us-east-1:644160558196:alarm:c7n-test-alarm-tags-filter', 'Dimensions': []}, 'supplementaryConfiguration': {}}
```

**Comment by kapilt** [comment]
re config, the last few prs on config support, i'm noticing there's pretty large variance on the number of important resource data they fail to capture. any issues with config data capture, are definitely something to take up directly with the service team via tam, its not something we can really handle. we can note those deficiets in the resource doc string is probably the best we can do, but its probably something that can age out.

**Comment by PratMis** [comment]
> re config, the last few prs on config support, i'm noticing there's pretty large variance on the number of important resource data they fail to capture. any issues with config data capture, are definitely something to take up directly with the service team via tam, its not something we can really handle. we can note those deficiets in the resource doc string is probably the best we can do, but its probably something that can age out.

Yup, making a note of these so that I can bring it up to our TAM. Thanks @kapilt

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**